redisvl/llmcache/semantic.py

```diff
@@ -6,56 +6,94 @@
 from redisvl.llmcache.base import BaseLLMCache
 from redisvl.providers import HuggingfaceProvider
 from redisvl.providers.base import BaseProvider
-from redisvl.query import create_vector_query
+from redisvl.query import VectorQuery
 from redisvl.utils.utils import array_to_buffer
 
 
 class SemanticCache(BaseLLMCache):
     """Cache for Large Language Models."""
 
+    # TODO allow for user to change default fields
     _default_fields = [
         VectorField(
             "prompt_vector",
             "FLAT",
             {"DIM": 768, "TYPE": "FLOAT32", "DISTANCE_METRIC": "COSINE"},
         ),
```

Comment:
right now it looks like the default fields is "fixed" but you can alter the embeddings provider. would this cause a potential clash with the schema (embedding dimensions etc)?

Reply:
There is a TODO in there about this. Struggling with the intersection between hand-holding and custom capability.